seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/TaskGroupImmutableInformation.java

```diff
@@ -40,6 +41,8 @@ public class TaskGroupImmutableInformation implements IdentifiedDataSerializable
 
     private Set<URL> jars;
 
+    private Set<ConnectorJarIdentifier> connectorJarIdentifiers;
```

Comment:
Why we need maintain two different jars collection? It looks bad.

Reply:
ConnectorJarIdentifiers represent the unique identifier of a Jar package file, which contains more information about the Jar package file, including the name of the connector plugin using the current Jar, the type of the current Jar package, and so on. If we use the Identifier to obtain Jar packages or distributed file systems (functions that have not yet been implemented) from the master node, it will be much more convenient because the root directory of Jar packages stored on the distributed file system can be configured by the user. We can generate a storage path on the distributed file system based on user user-configured root directory, Jar package names, Jar package types, and other information, achieving more flexible storage. If we want the Jar package storage path on the distributed file system to be completely consistent with the local path, then I can remove it.

Reply:
cc @hailin0 @EricJoy2048 WDYT?